### PR TITLE
python3Packages.adafruit* and python3Packages.binho-host-adapter: init at 0.1.6

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
All three are dependencies for other Adafruit Python modules and perhaps for CircuitPython. 

Those are dependencies of dependencies which are used by Home Assistant.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 110378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.adafruit-platformdetect</li>
    <li>python37Packages.adafruit-pureio</li>
    <li>python37Packages.binho-host-adapter</li>
    <li>python38Packages.adafruit-platformdetect</li>
    <li>python38Packages.adafruit-pureio</li>
    <li>python38Packages.binho-host-adapter</li>
    <li>python39Packages.adafruit-platformdetect</li>
    <li>python39Packages.adafruit-pureio</li>
    <li>python39Packages.binho-host-adapter</li>
  </ul>
</details>

File: pkgs/development/python-modules/adafruit-platformdetect/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+buildPythonPackage rec {
+  pname = "Adafruit-PlatformDetect";
+  version = "2.27.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0rnmy74rjjcyni5sr8h1djffpj7wngn2wqckl5vknp2smaihp34l";
+  };
+
+  nativeBuildInputs = [ setuptools_scm ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/adafruit-platformdetect/default.nix:16]:
```suggestion
  nativeBuildInputs = [ setuptools-scm ];
```

File: pkgs/development/python-modules/adafruit-pureio/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "Adafruit-PureIO";
+  version = "1.1.8";
+
+  src = fetchPypi {
+    pname = "Adafruit_PureIO";
+    inherit version;
+    sha256 = "1mfa6sfz7qwgajz3lqw0s69ivvwbwvblwkjzbrwdrxjbma4jaw66";
+  };
+
+  nativeBuildInputs = [ setuptools_scm ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/adafruit-pureio/default.nix:17]:
```suggestion
  nativeBuildInputs = [ setuptools-scm ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.adafruit-platformdetect</li>
<li>python37Packages.adafruit-pureio</li>
<li>python37Packages.binho-host-adapter</li>
<li>python38Packages.adafruit-platformdetect</li>
<li>python38Packages.adafruit-pureio</li>
<li>python38Packages.binho-host-adapter</li>
<li>python39Packages.adafruit-platformdetect</li>
<li>python39Packages.adafruit-pureio</li>
<li>python39Packages.binho-host-adapter</li>
</ul>
</details>

File: pkgs/development/python-modules/adafruit-platformdetect/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/adafruit-platformdetect/default.nix]: Thanks, updated.

File: pkgs/development/python-modules/adafruit-pureio/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/adafruit-pureio/default.nix]: Upsated

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 110378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.adafruit-platformdetect</li>
    <li>python37Packages.adafruit-pureio</li>
    <li>python37Packages.binho-host-adapter</li>
    <li>python38Packages.adafruit-platformdetect</li>
    <li>python38Packages.adafruit-pureio</li>
    <li>python38Packages.binho-host-adapter</li>
    <li>python39Packages.adafruit-platformdetect</li>
    <li>python39Packages.adafruit-pureio</li>
    <li>python39Packages.binho-host-adapter</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110378` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.adafruit-platformdetect</li>
<li>python37Packages.adafruit-pureio</li>
<li>python37Packages.binho-host-adapter</li>
<li>python38Packages.adafruit-platformdetect</li>
<li>python38Packages.adafruit-pureio</li>
<li>python38Packages.binho-host-adapter</li>
<li>python39Packages.adafruit-platformdetect</li>
<li>python39Packages.adafruit-pureio</li>
<li>python39Packages.binho-host-adapter</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110378` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.adafruit-platformdetect</li>
<li>python37Packages.adafruit-pureio</li>
<li>python37Packages.binho-host-adapter</li>
<li>python38Packages.adafruit-platformdetect</li>
<li>python38Packages.adafruit-pureio</li>
<li>python38Packages.binho-host-adapter</li>
<li>python39Packages.adafruit-platformdetect</li>
<li>python39Packages.adafruit-pureio</li>
<li>python39Packages.binho-host-adapter</li>
</ul>
</details>

**Pull request merged**